### PR TITLE
feat: add support for React 18

### PR DESCRIPTION
**Pull request by MartinCupela**
### 🎯 Goal

Upgrade to React v18.

depends on https://github.com/GetStream/react-file-utils/pull/58

### 🛠 Implementation details

1. Upgrade `react` +  `react-dom` dependencies
2. Upgrade testing libraries
4. Explicitly declare `children?: React.ReactNode` in component props type definitions or use `PropsWithChildren`
5. Explicitly declare the return value of translation function provided by `Stream18n`
6. Unify datetime values formatting

### 🎨 UI Changes

_Add relevant screenshots_

### CODE REVIEW

File: src/components/Channel/Channel.tsx [hunk 1]
```diff
@@ -850,7 +850,7 @@ const ChannelInner = <
       ? 'str-chat--windows-flags'
       : '';
 
-  const NullProvider: React.FC = ({ children }) => <>{children}</>;
+  const NullProvider: React.FC<{ children?: React.ReactNode }> = ({ children }) => <>{children}</>;
```
Comment (arnautov-anton) [inline src/components/Channel/Channel.tsx:853]:
```suggestion
  const NullProvider = React.Fragment;
```

**Comment by MAK-Denis** [comment]
I'm waiting for this update. Please notify here when it's done. Appreciate for all your hard work!!!

File: src/components/Channel/hooks/useIsMounted.ts [hunk 2]
```diff
@@ -3,12 +3,12 @@ import { useEffect, useRef } from 'react';
 export const useIsMounted = () => {
   const isMounted = useRef(true);
```
Comment (arnautov-anton) [inline src/components/Channel/hooks/useIsMounted.ts:4]:
```suggestion
  const isMounted = useRef(false);
```

**Review by petyosi — APPROVED** [review]

**Comment by petyosi** [comment]
I solved the React 18 problem in react-virtuoso in v2.10.2, so no it works with or without the property set. We can remove the notice from the docs and upgrade the version in package.json.

File: src/components/AutoCompleteTextarea/Textarea.jsx [hunk 3]
```diff
@@ -99,16 +99,6 @@ export class ReactTextareaAutocomplete extends React.Component {
     if (!this.textareaRef) return;
 
     const trigger = this.state.currentTrigger;
-    const hasFocus = this.textareaRef.matches(':focus');
```
[the remark this answers is not in the document]
Comment (arnautov-anton) [reply src/components/AutoCompleteTextarea/Textarea.jsx:137]: It does not make sense to keep this check anymore - `onKeyDown` event is fired only when the element on which this handler is registered is being actively focused. This check is there due to previous implementation of this handler which was registered globally on the `window` object.

File: src/components/MessageInput/__tests__/MessageInput.test.js [hunk 4]
```diff
@@ -737,123 +737,90 @@ function axeNoViolations(container) {
...
 
+        act(() => fireEvent.keyDown(input, { key: 'Enter' }));
+
+        expect(submitHandler).not.toHaveBeenCalled();
         await waitFor(axeNoViolations(container));
       });
 
-      it('should submit if [[16,13], [57], [48]] are provided as keycodeSubmitKeys and keydown events match 57', async () => {
-        const { container, findByPlaceholderText, submit } = renderComponent({
+      it('should submit if shouldSubmit function is provided and keydown events do match', async () => {
```
[the remark this answers is not in the document]
Comment (arnautov-anton) [reply src/components/MessageInput/__tests__/MessageInput.test.js:765]: Good point, will add. :+1:

**Review by arnautov-anton — APPROVED** [review]

**Review by arnautov-anton — DISMISSED** [review]

**Review by arnautov-anton — APPROVED** [review]

**Pull request merged**